### ISSUE TITLE
is:issue is:open [swagger-zod] `readOnly: true` generates with `.readonly` when they are not equivalent

### ISSUE THREAD

**Issue by mitchell-merry** [kubb-project/kubb#857]
### What version of `kubb` is running?

kubb/2.7.1 darwin-arm64 node-v18.19.1

### What platform is your computer?

MacOS

### What version of external packages are you using(`@tanstack-query`, `MSW`, `React`, `Vue`, ...)

nothing related

### What steps can reproduce the bug?

I believe the current behaviour is intended. When a schema has `readOnly: true`, the zod schemas generate with `.readonly()`.

### How often does this bug happen?

Every time

### What is the expected behavior?

`readonly: true` only means that the property is not included in write requests (as implemented in https://github.com/kubb-project/kubb/pull/495/files). It should not apply `.readonly()` - it makes no comment on the mutability of the object in the application.

### Swagger/OpenAPI file?

```yaml
swagger: "2.0"

info:
  title: Some API
  version: 2.0.0

paths:
  /thing:
    get:
      operationId: getThing
      parameters: []
      responses:
        200:
          description: Some response
          schema:
            type: object
            properties:
              foo:
                type: number
                readOnly: true
```

which generates:
```ts
import { z } from 'zod';

/**
 * @description Some response
 */
export const getThing200Schema = z.object({
  foo: z.number().readonly().optional(),
});

/**
 * @description Some response
 */
export const getThingQueryResponseSchema = z.object({
  foo: z.number().readonly().optional(),
});
```

**Comment by stijnvanhulle** [comment]
Fixed with Kubb `v2.8.1`

**Comment by mitchell-merry** [comment]
Thank you!
